### PR TITLE
refactor: 1816 Rename Data Normaliser

### LINKED ISSUE #1816: Appropriate naming for  DataNormaliserND classes
Rename `DataNormaliserND` classes to `DataOperatorND` for better clarity. The current naming doesn't accurately reflect their primary function - operating on data, rather than normalising data; this would include the specific normalisation operations (which currently exist) as well as generic data operations such as `divide()` and `multiply()`.

### PR DESCRIPTION
**Pull request by jws-1**
Small PR renaming `DataNormaliserND` to `DataOperatorND` (this also applies to the subclasses), also adds some additional comments to the relevant classes for clarity.

Closes #1816.

### CODE REVIEW

File: src/analyser/dataOperator3D.cpp [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 
-void DataNormaliser3D::normalise(NormalisationFunction3D normalisationFunction)
+/*
+ * Data Operation Functions
+ */
+
+// Generic operate function
+void DataOperator3D::operate(OperateFunction3D operateFunction)
 {
     Messenger::warn("Normalisation function not implemented for 3D data.");
```
Comment (trisyoungs) [inline src/analyser/dataOperator3D.cpp]:
```suggestion
    Messenger::warn("Data operate() function not implemented for 3D data.");
```

File: src/analyser/dataOperatorBase.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    /*
+     * Normalisation Functions
+     */
+    public:
+    // Perform grid normalisation
+    virtual void normaliseByGrid() = 0;
+    // Perform spherical shell normalisation
+    virtual void normaliseBySphericalShell() = 0;
+    // Normalise the target data to a given value
+    virtual void normaliseTo(double value = 1.0, bool absolute = true) = 0;
```
Comment (trisyoungs) [inline src/analyser/dataOperatorBase.h]: For ultraclarity I think I would suggest renaming this one:
```suggestion
    // Normalise the sum of the target data to a given value
    virtual void normaliseSumTo(double value = 1.0, bool absolute = true) = 0;
```

File: src/analyser/dataOperator1D.cpp [hunk 3]
```diff
@@ -1,26 +1,35 @@
...
+void DataOperator1D::operate(OperateFunction1D operateFunction)
 {
     const auto &xs = targetData_.xAxis();
     auto &values = targetData_.values();
 
-    const auto xDelta = xs.size() > 1 ? xs[1] - xs[0] : 1.0;
-
     for (auto i = 0; i < xs.size(); ++i)
-        values.at(i) = normalisationFunction(xs[i], xDelta, values.at(i));
+        values.at(i) = operateFunction(xs[i], targetData_.error(i), values.at(i));
```
Comment (trisyoungs) [inline src/analyser/dataOperator1D.cpp]: `targetData_.error(i)`?

**Review by trisyoungs — CHANGES_REQUESTED** [review]

File: src/analyser/dataOperator1D.cpp [hunk 4]
```diff
@@ -1,26 +1,36 @@
...
-DataNormaliser1D::DataNormaliser1D(Data1D &targetData) : DataNormaliserBase<Data1D, NormalisationFunction1D>(targetData) {}
+DataOperator1D::DataOperator1D(Data1D &targetData) : DataOperatorBase<Data1D, OperateFunction1D>(targetData) {}
 
-void DataNormaliser1D::normalise(NormalisationFunction1D normalisationFunction)
+/*
+ * Data Operation Functions
+ */
+
+// Generic operate function
+void DataOperator1D::operate(OperateFunction1D operateFunction)
```
Comment (trisyoungs) [inline src/analyser/dataOperator1D.cpp:15]: I think just calling the function argument `operator` feels a lot cleaner:
```suggestion
void DataOperator1D::operate(OperateFunction1D operator)
```

File: src/modules/epsr/process.cpp [hunk 5]
```diff
@@ -424,7 +424,7 @@ Module::ExecutionResult EPSRModule::process(ModuleContext &moduleContext)
 
             // Always add absolute data to the scattering matrix - if the calculated data has been normalised, remove this
             // normalisation from the reference data (we assume that the two are consistent)
-            auto normType = module->keywords().getEnumeration<StructureFactors::NormalisationType>("NormaliseTo");
+            auto normType = module->keywords().getEnumeration<StructureFactors::NormalisationType>("normaliseTo");
```
Comment (trisyoungs) [inline src/modules/epsr/process.cpp]: Bit of aggressive search/replacing!
```suggestion
            auto normType = module->keywords().getEnumeration<StructureFactors::NormalisationType>("NormaliseTo");
```

File: src/modules/epsr/process.cpp [hunk 6]
```diff
@@ -447,7 +447,7 @@ Module::ExecutionResult EPSRModule::process(ModuleContext &moduleContext)
             auto refMinusIntra = originalReferenceData;
             Interpolator::addInterpolated(weightedSQ.boundTotal(), refMinusIntra, -1.0);
 
-            auto normType = module->keywords().getEnumeration<StructureFactors::NormalisationType>("NormaliseTo");
+            auto normType = module->keywords().getEnumeration<StructureFactors::NormalisationType>("normaliseTo");
```
Comment (trisyoungs) [inline src/modules/epsr/process.cpp]:
```suggestion
            auto normType = module->keywords().getEnumeration<StructureFactors::NormalisationType>("NormaliseTo");
```

**Review by trisyoungs — APPROVED** [review]
Looks good.  Just a couple comments on naming.

File: src/modules/epsr/process.cpp (hunk 5 above)
Comment (jws-1) [reply src/modules/epsr/process.cpp]: Agh!...

**Comment by trisyoungs** [comment]
@jws-1 The suggestion to rename to `operator` may be the worst code suggestion I've ever made - sorry about that!  Silly reserved words....

**Comment by jws-1** [comment]
> @jws-1 The suggestion to rename to `operator` may be the worst code suggestion I've ever made - sorry about that! Silly reserved words....

Hah. I changed it and then forgot about it! I'll revert.

**Pull request merged**